.eslintrc.cjs

```diff
@@ -0,0 +1,108 @@
+module.exports = {
+  env: {
+    browser: true,
+    es2024: true,
+  },
+  extends: [
+    'plugin:react/recommended',
+    "plugin:react-hooks/recommended",
+    'airbnb-typescript',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:prettier/recommended',
+    'plugin:cypress/recommended',
+  ],
+  overrides: [
+    {
+      'files': ['**/*.spec.jsx'],
+      'rules': {
+        'react/jsx-filename-extension': ['off'],
+      }
+    }
+  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaFeatures: {
+      jsx: true,
+    },
+    ecmaVersion: 12,
+    project: './tsconfig.json',
+    sourceType: 'module',
+  },
+  plugins: [
+    'jsx-a11y',
+    'import',
+    'react-hooks',
+    '@typescript-eslint',
+    'prettier'
+  ],
+  rules: {
+    // JS
+    'semi': 'off',
+    '@typescript-eslint/semi': ['error', 'always'],
+    'prefer-const': 2,
+    curly: [2, 'all'],
+    'max-len': ['error', {
+      ignoreTemplateLiterals: true,
+      ignoreComments: true,
+    }],
+    'no-redeclare': [2, { builtinGlobals: true }],
+    'no-console': 2,
+    'operator-linebreak': 0,
+    'brace-style': [2, '1tbs'],
+    'arrow-body-style': 0,
+    'arrow-parens': 0,
+    'no-param-reassign': [2, { props: true }],
+    'padding-line-between-statements': [
+      2,
+      { blankLine: 'always', prev: '*', next: 'return' },
+      { blankLine: 'always', prev: ['const', 'let', 'var'], next: '*' },
+      { blankLine: 'any', prev: ['const', 'let', 'var'], next: ['const', 'let', 'var'] },
+      { blankLine: 'always', prev: 'directive', next: '*' },
+      { blankLine: 'always', prev: 'block-like', next: '*' },
+    ],
+    'implicit-arrow-linebreak:': 0,
```

Comment:
There's a typo in the rule `implicit-arrow-linebreak:`. The colon at the end should be removed to correctly disable this rule.